itest/lnd_signer_test.go

```diff
@@ -458,4 +460,63 @@ func runSignVerifyMessage(ht *lntest.HarnessTest, alice *node.HarnessNode) {
 	}
 	verifyResp = alice.RPC.VerifyMessageSigner(verifyReq)
 	require.True(ht, verifyResp.Valid, "failed to verify message")
+
+	// Now let's try signing and verifying a tagged hash.
+	msg, err := hex.DecodeString("932b9ca9cee9172ee1ad2944401dd94b99824f" +
+		"a5155193bc6743d91319e70fff")
+	require.NoError(ht, err)
+	tag := []byte("lightninginvoice_requestsignature")
+
+	signMsgReq = &signrpc.SignMessageReq{
+		Msg:        msg,
+		KeyLoc:     keyLoc,
+		SchnorrSig: true,
+		Tag:        tag,
+	}
+	signMsgResp = alice.RPC.SignMessageSigner(signMsgReq)
+	customPubKey = deriveCustomizedKey()
+
+	verifyReq = &signrpc.VerifyMessageReq{
+		Msg:          msg,
+		Signature:    signMsgResp.Signature,
+		Pubkey:       schnorr.SerializePubKey(customPubKey),
+		IsSchnorrSig: true,
+		Tag:          tag,
+	}
+	verifyResp = alice.RPC.VerifyMessageSigner(verifyReq)
+	require.True(ht, verifyResp.Valid, "failed to verify message")
+
+	// Verify that both SignMessage and VerifyMessage error if a tag is
+	// provided but the Schnorr option is not set.
+	signMsgReq = &signrpc.SignMessageReq{
+		Msg:    msg,
+		KeyLoc: keyLoc,
+		Tag:    tag,
+	}
+
+	ctxt := context.Background()
+	_, err = alice.RPC.Signer.SignMessage(ctxt, signMsgReq)
+	require.Error(ht, err)
```

Comment:
should use `require.ErrorContains` here and below to assert that we are getting the error we expect and not just any error